package proto_ff;

// calm 自定义的fieldoption都在该文件中
import "yd_fieldoptions.proto";
import "Com.proto";
import "ComDefine.proto";
import "DBProxy.proto";

//用来测试db的
message dbServerMgr
{
  optional uint64 id			= 1 [(yd_fieldoptions.db_field_type) = E_FIELDTYPE_PRIMARYKEY, (yd_fieldoptions.db_field_comment) = "ID", (yd_fieldoptions.db_field_auto_increment) = true, (yd_fieldoptions.db_field_auto_increment_value) = 10000];
  optional string contract = 2 [(yd_fieldoptions.db_field_type) = E_FIELDTYPE_UNIQUE_INDEX, (yd_fieldoptions.db_field_bufsize)=128, (yd_fieldoptions.db_field_comment) = "合同号"];
  optional string machine_addr = 3 [(yd_fieldoptions.db_field_type) = E_FIELDTYPE_INDEX, (yd_fieldoptions.db_field_bufsize)=128, (yd_fieldoptions.db_field_comment) = "机器码"];
  optional string ip = 4 [(yd_fieldoptions.db_field_type) = E_FIELDTYPE_INDEX, (yd_fieldoptions.db_field_bufsize)=128, (yd_fieldoptions.db_field_comment) = "IP地址"];
  optional string bus_name = 5 [(yd_fieldoptions.db_field_type) = E_FIELDTYPE_INDEX, (yd_fieldoptions.db_field_bufsize)=128, (yd_fieldoptions.db_field_comment) = "服务器ID"];
  optional string server_desc = 6 [(yd_fieldoptions.db_field_bufsize)=128, (yd_fieldoptions.db_field_comment) = "描述信息"];
  optional uint32 cur_count = 7 [(yd_fieldoptions.db_field_comment) = "当前服务器数"];
  optional uint64 last_login_time = 8 [(yd_fieldoptions.db_field_comment) = "上一次登录时间"];
  optional uint64 last_logout_time = 9 [(yd_fieldoptions.db_field_comment) = "上一次下线时间"];
  repeated uint64 logout_time = 10 [(yd_fieldoptions.db_field_arysize)=3];
  repeated string desc = 11 [(yd_fieldoptions.db_field_bufsize)=128, (yd_fieldoptions.db_field_arysize)=2];
  repeated ComPair pair = 12 [(yd_fieldoptions.db_field_arysize)=2];
  repeated BestEQSlotInfo facade = 13 [(yd_fieldoptions.db_field_arysize)=2];
}

// 角色DB数据
message GetRegisterNum_RoleDBData
{
  optional uint32 num = 1;
}

message RoleDBSnsSimple
{
  optional uint64                 cid					= 1[(yd_fieldoptions.db_field_type) = E_FIELDTYPE_PRIMARYKEY];	// role cid
  optional uint32                 zid					= 2;    // 区服ID
  optional uint32					        uid					= 3;    // UID
  optional RoleDBBaseData         base				= 4;    // 基础数据
}

message MyFriendInfo
{

}

message RoleDBSnsDetail
{
  optional uint64               cid	= 1[(yd_fieldoptions.db_field_type) = E_FIELDTYPE_PRIMARYKEY];	// role cid
  optional MyFriendInfo         friend_info = 2;
}

message RoleEnterSceneData
{
  optional uint64                 cid					= 1;	// role cid
  optional uint32                 zid					= 2;    // 区服ID
  optional uint32					        uid					= 3;    // UID
  optional RoleDBBaseData         base				= 10;    // 基础数据
  optional AttrDBData             attr        = 11;
}

message RoleDBName
{
  optional bytes name = 1[(yd_fieldoptions.db_field_type) = E_FIELDTYPE_PRIMARYKEY, (yd_fieldoptions.field_bufsize)=32, (yd_fieldoptions.db_field_bufsize)=32, (yd_fieldoptions.db_field_comment) = "角色名字"];	// role cid
}